kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/ScanSuite.scala

```diff
@@ -1594,6 +1595,125 @@ class ScanSuite extends AnyFunSuite with TestUtils with ExpressionTestUtils with
       )
     }
   }
+
```

Comment:
Can you combine the common code for all these tests please? You can use test helper functions (there should be examples of these throughout the test code)